src/modules/universal-discovery/components/content-meta-preview/content.meta.preview.component.js

```diff
@@ -13,16 +14,51 @@ export default class ContentMetaPreviewComponent extends Component {
             imageUri: '',
             translations: [],
             selectContentEnabled: false,
+            bookmarked: null,
         };
 
+        this.removeBookmark = this.removeBookmark.bind(this);
+        this.addBookmark = this.addBookmark.bind(this);
         this.toggleEnabledState = this.toggleEnabledState.bind(this);
+        this.setBookmarked = this.setBookmarked.bind(this);
     }
 
     componentWillReceiveProps(props) {
+        const { id } = props.data;
         const translations = this.getTranslations(props.data);
         const imageUri = this.getImageUri(props.data);
 
-        this.setState((state) => ({ ...state, translations, imageUri }));
+        this.setState((state) => ({ ...state, translations, imageUri, bookmarked: null }));
+
+        this.checkIfBookmarked(id);
```

Comment:
I still don't like making the request on every selected content change. Can we think about getting the whole list at once and check in this list?